Src/AutoMoq/AutoMockPropertiesCommand.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Moq;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.AutoMoq
+{
+    /// <summary>
+    /// A command that populates all public writable properties/fields of a mock object with anonymous values.
+    /// </summary>
+    public class AutoMockPropertiesCommand : ISpecimenCommand
```

Comment:
It's a thin layer that decorates/builds on top of `AutoPropertiesCommand`.

It "unwraps" the mock's `Object` property, and adds a stricter request specification to filter out dynamic proxy fields.

I think the alternative would be to:
 * Delete the `AutoMockPropertiesCommand` class
 * Make the `IgnoreProxyMembersSpecification` specification public and non-nested.
 * And change the composition root from this:

```csharp
            fixture.Customizations.Add(
                new Postprocessor(
                    builder: new MockPostprocessor(
                                new MethodInvoker(
                                    new MockConstructorQuery())),
                    command: new CompositeSpecimenCommand(
                                new MockVirtualMethodsCommand(),
                                new StubPropertiesCommand(),
                                new AutoMockPropertiesCommand())));

            fixture.ResidueCollectors.Add(Relay);
```

to this:

```csharp

            fixture.Customizations.Add(
                new Postprocessor(
                    builder: new MockPostprocessor(
                                new MethodInvoker(
                                    new MockConstructorQuery())),
                    command: new CompositeSpecimenCommand(
                                new MockVirtualMethodsCommand(),
                                new StubPropertiesCommand())));

            fixture.ResidueCollectors.Add(
                new Postprocessor(
                    Relay,
                    new AutoPropertiesCommand(
                        new IgnoreProxyMembersSpecification())));
```

Personally, I prefer the way it's currently implemented. `IgnoreProxyMembersSpecification` is not really intuitive, and is too specific to ever be reused (I think), so it's probably best to make it nested and private. `AutoMockPropertiesCommand` is an intuitive layer (albeit thin) and a well-defined responsibility. And the composition root looks cleaner and is easier to understand.